tox.ini

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = lint, unit
+
+[vars]
+src_path = {toxinidir}/src/
+tst_path = {toxinidir}/tests/
+;lib_path = {toxinidir}/lib/charms/operator_name_with_underscores
+all_path = {[vars]src_path} {[vars]tst_path} 
+
+[testenv]
+setenv =
+  PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
+  PYTHONBREAKPOINT=ipdb.set_trace
+  PY_COLORS=1
+passenv =
+  PYTHONPATH
+  CHARM_BUILD_DIR
+  MODEL_SETTINGS
+
+[testenv:fmt]
+description = Apply coding style standards to code
+deps =
+    black
+    isort
+commands =
+    isort {[vars]all_path}
+    black {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+deps =
+    black
```

Comment:
nit: could just throw these in the test-requirements.txt rather than inline some and not others. Its kind of nice if there's generally one or two places to go and its consistent. Gets to be an awkward contributor experience if you have really dive into the different ways dependencies are managed and defined.

Reply:
Just a comment here: in general I've been encouraging charm teams to have all runtime dependencies in requirements.txt, and all dev/test dependencies in the respective environment in tox.ini - but either way if test-requirements.txt is knocking around we could kill it in favour of this.

Reply:
I  kept the dev dependencies in tox.ini as we talked in the meetings today.

Reply:
I think this is reasonable and really up to the development team to make this decision. As a note, the openstack charms (of which I have most familiarity) do it by dropping requirements into various {test-,build-,}requirements.txt files and we occasionally sync centralized copies of these files across all charms to have standard dependencies across all openstack charm projects.